src/Microsoft.AspNetCore.SignalR.Protocols.MsgPack/Internal/Protocol/MessagePackHubProtocol.cs

```diff
@@ -233,22 +233,15 @@ private static object[] BindArguments(Unpacker unpacker, IReadOnlyList<Type> par
 
         public void WriteMessage(HubMessage message, Stream output)
```

Comment:
@anurse and I spoke about this a while back and we were saying that it would be interesting to look at a fixed(ish) length prefix so we could write directly to the output stream but also reserve a fixed chunk at the beginning to write the length. We would avoid the double stream allocation.

Reply:
Would that work when writing to a pipe?

Reply:
It should. The buffer you get back shouldn't move until you flush to the other side.

Reply:
Lets track that separately. I have thoughts, but I wouldn't want to bog down this PR. A fully-fixed length field would be best, but we'd probably have to use 8 bytes for that and that's a lot of length for messages that will usually be less than 2^16 bytes. We could set a limit of 2^16, but we don't really want an arbitrary limit that low. My thought was to do something with segmenting the message, and build a stream that wrote out 2^15 byte chunks, with a 2-byte prefix (length + a bit indicating if there is another chunk following). That way the MsgPack serializer doesn't have to cooperate with us, it can just write to a stream. Now that we're using an array directly though, I'm not sure it would work as well...